.github/workflows/publish_images_on_push.yaml

```diff
@@ -0,0 +1,81 @@
+# (C) Copyright Confidential Containers Contributors 2023.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Build and push project images on pull requests merged.
+---
+name: Publish project images
+
+on:
+  push:
+    branches:
+      - 'main'
+
+  workflow_dispatch:
+
+jobs:
+  get_changed_files:
+    runs-on: ubuntu-latest
+    outputs:
+      changed_files: ${{ steps.changed-files.outputs.all_changed_files }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v40
+
+  caa:
+    uses: ./.github/workflows/caa_build_and_push_per_arch.yaml
+    with:
+      git_ref: ${{ github.sha }}
+    secrets: inherit
+
+  csi-wrapper:
+    needs: [get_changed_files]
+    uses: ./.github/workflows/csi_wrapper_images.yaml
+    if: >
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'push' &&
+      contains(needs.get_changed_files.outputs.changed_files, 'volumes/csi-wrapper/'))
+    with:
+      image_tags: latest,${{ github.sha }}
+      git_ref: ${{ github.sha }}
+    secrets: inherit
+
+  peerpod-ctrl:
+    needs: [get_changed_files]
+    uses: ./.github/workflows/peerpod-ctrl_image.yaml
+    if: >
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'push' &&
+      contains(needs.get_changed_files.outputs.changed_files, 'peerpod-ctrl/'))
```

Comment:
that looks pretty elegant, but will this be sufficient to cover changes that should trigger a rebuild? if changes have occured e.g. in `pkg/*` we'd also want an updated build, no?